src/artemis/external_interaction/tests/conftest.py

```diff
@@ -0,0 +1,125 @@
+from unittest.mock import patch
+
+import pytest
+
+from artemis.parameters import ISPYB_PLAN_NAME
+
+
+@pytest.fixture
+def nexus_writer():
+    with patch("artemis.external_interaction.communicator_callbacks.NexusWriter") as nw:
+        yield nw
+
+
+@pytest.fixture
+def run_start():
+    with patch("artemis.external_interaction.communicator_callbacks.run_start") as p:
+        yield p
+
+
+@pytest.fixture
+def run_end():
+    with patch("artemis.external_interaction.communicator_callbacks.run_end") as p:
+        yield p
+
+
+@pytest.fixture
+def wait_for_result():
+    with patch(
+        "artemis.external_interaction.communicator_callbacks.wait_for_result"
+    ) as wfr:
+        yield wfr
+
+
+@pytest.fixture
+def mock_ispyb_get_time():
+    with patch(
+        "artemis.external_interaction.communicator_callbacks.StoreInIspyb3D.get_current_time_string"
+    ) as p:
+        yield p
+
+
+@pytest.fixture
+def mock_ispyb_store_grid_scan():
+    with patch(
+        "artemis.external_interaction.communicator_callbacks.StoreInIspyb3D.store_grid_scan"
+    ) as p:
+        yield p
+
+
+@pytest.fixture
+def mock_ispyb_update_time_and_status():
+    with patch(
+        "artemis.external_interaction.communicator_callbacks.StoreInIspyb3D.update_grid_scan_with_end_time_and_status"
+    ) as p:
+        yield p
+
+
+@pytest.fixture
+def mock_ispyb_begin_deposition():
+    with patch(
+        "artemis.external_interaction.communicator_callbacks.StoreInIspyb3D.begin_deposition"
+    ) as p:
+        yield p
+
+
+@pytest.fixture
+def mock_ispyb_end_deposition():
+    with patch(
+        "artemis.external_interaction.communicator_callbacks.StoreInIspyb3D.end_deposition"
+    ) as p:
+        yield p
```

Comment:
Nit: Having all these fixtures seems messy to me but I can see that the alternative is to have lots of patches everywhere, which is equally messy. Not sure of a good solution though, maybe look into/we can discuss?

Reply:
that's why they're off in their own conftest file where they don't bother you when reading the tests... best solution I could come up with

Reply:
Reluctantly agree